### PR TITLE
[openfga] Configure CloudSQL datastore

### PR DESCRIPTION
**Pull request by easyCZ**
## Related Issue(s)

* Depends on https://github.com/gitpod-io/gitpod/pull/15692

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

File: install/installer/pkg/components/openfga/deployment.go [hunk 1]
```diff
@@ -28,75 +28,10 @@ func deployment(ctx *common.RenderContext) ([]runtime.Object, error) {
 		return nil, nil
 	}
 
-	containers := []corev1.Container{
-		{
```
Comment (easyCZ) [inline install/installer/pkg/components/openfga/deployment.go:32]: The definition is moved below. If there's config for CloudSQL, we configure the sidecar first and based on that we inject extra envs to the openfga application.

File: install/installer/pkg/components/openfga/deployment.go [hunk 2]
```diff
@@ -137,8 +72,108 @@ func deployment(ctx *common.RenderContext) ([]runtime.Object, error) {
...
+					},
+					Key: "user",
+				}},
+			},
+			{
+				Name:  "OPENFGA_DATASTORE_URI",
+				Value: fmt.Sprintf("$(DB_USERNAME):$(DB_PASSWORD)@tcp(%s:%d)/%s?parseTime=true", dbHost, CloudSQLProxyPort, cfg.CloudSQL.Instance),
+			},
+		}...)
+	}
```
Comment (easyCZ) [inline install/installer/pkg/components/openfga/deployment.go:106]: This is the core of this PRs change, rest are moves to satisfy declaration dependencies.

**Comment by easyCZ** [comment]
Cloud sql has successfully deployed in staging, we can proceed with this

/unhold

File: install/installer/pkg/components/openfga/deployment.go [hunk 3]
```diff
@@ -137,8 +72,108 @@ func deployment(ctx *common.RenderContext) ([]runtime.Object, error) {
...
+				Name:  "OPENFGA_DATASTORE_ENGINE",
+				Value: "mysql",
+			},
+			{
+				Name: "DB_PASSWORD",
+				ValueFrom: &corev1.EnvVarSource{SecretKeyRef: &corev1.SecretKeySelector{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: cfg.CloudSQL.DatabaseSecretRef,
+					},
+					Key: "password",
```
Comment (andrew-farries) [inline install/installer/pkg/components/openfga/deployment.go:89]: How will the real username and password be provided to the env var?

**Review by andrew-farries — APPROVED** [review]

File: install/installer/pkg/components/openfga/deployment.go (hunk 3 above)
Comment (easyCZ) [reply install/installer/pkg/components/openfga/deployment.go:89]: Through a k8s secret, which has been added in this PR https://github.com/gitpod-io/ops/pull/7712

TF loads the secret from GCP into k8s, we reference it through the `cfg.CloudSQL.DatabaseSecretRef` config

**Pull request merged**